Framework/Kernel/inc/MantidKernel/ListValidator.h

```diff
@@ -76,10 +76,39 @@ template <typename TYPE> class ListValidator : public TypedValidator<TYPE> {
   }
 
   /** Constructor
-   *  @param values :: A vector of the valid values
+   *  @param values :: An array of the valid values
    *  @param aliases :: Optional aliases for the valid values.
    *  @param allowMultiSelection :: True if the list allows multi selection
    */
+  template <std::size_t SIZE>
+  explicit ListValidator(const std::array<TYPE, SIZE> &values,
```

Comment:
@josephframsay This is a nice improvement. The only thing I'm not too keen on is the code duplication in the constructors.

We could swap the template here for `template<typename T>` and then delete the second `vector`-based constructor. As we're using iterators then this should work for all of the container types.